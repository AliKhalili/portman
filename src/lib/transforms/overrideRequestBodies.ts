/* eslint-disable @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any */
export const overrideRequestBodies = (obj: any): any => {
  obj.item.map(resource => {
    const create = resource.item.find(({ request: { method } }) => {
      return method === 'POST'
    })

    const update = resource.item.find(({ request: { method } }) => {
      return method === 'PATCH'
    })

    if (update && create) {
      update.request.body = { ...create.request.body }
    }
  })
  return obj
}
