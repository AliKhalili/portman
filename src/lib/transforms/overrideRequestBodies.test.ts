import { overrideRequestBodies } from './overrideRequestBodies'

describe('overrideRequestBodies()', () => {
  const createBody = { foo: 'bar', baz: '{{buzz}}' }
  const updateBody = { foo: 'bar', baz: 'buzz' }

  it('should replace UPDATE requestBody with one from CREATE', () => {
    const obj = {
      item: [
        {
          name: 'RequestFolder',
          item: [
            {
              request: {
                method: 'POST',
                body: createBody
              }
            },
            {
              request: {
                method: 'PATCH',
                body: updateBody
              }
            }
          ]
        }
      ]
    }

    const transform = overrideRequestBodies(obj)
    expect(transform).toMatchSnapshot()
  })

  it('should return unaltered if there is no update method', () => {
    const obj = {
      item: [
        {
          name: 'RequestFolder',
          item: [
            {
              request: {
                method: 'POST',
                body: createBody
              }
            }
          ]
        }
      ]
    }

    const transform = overrideRequestBodies(obj)
    expect(transform).toMatchSnapshot()
  })
})
