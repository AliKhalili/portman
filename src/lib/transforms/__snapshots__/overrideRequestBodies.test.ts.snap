// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overrideRequestBodies() should replace UPDATE requestBody with one from CREATE 1`] = `
Object {
  "item": Array [
    Object {
      "item": Array [
        Object {
          "request": Object {
            "body": Object {
              "baz": "{{buzz}}",
              "foo": "bar",
            },
            "method": "POST",
          },
        },
        Object {
          "request": Object {
            "body": Object {
              "baz": "{{buzz}}",
              "foo": "bar",
            },
            "method": "PATCH",
          },
        },
      ],
      "name": "RequestFolder",
    },
  ],
}
`;

exports[`overrideRequestBodies() should return unaltered if there is no update method 1`] = `
Object {
  "item": Array [
    Object {
      "item": Array [
        Object {
          "request": Object {
            "body": Object {
              "baz": "{{buzz}}",
              "foo": "bar",
            },
            "method": "POST",
          },
        },
      ],
      "name": "RequestFolder",
    },
  ],
}
`;
